### PR TITLE
advanced settings implementation

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: src/app/shared/components/qbd/configuration/advanced-setting/advanced-setting.component.ts [hunk 1]
```diff
@@ -1,15 +1,141 @@
...
+        expenseMemoStructure: [this.advancedSettings?.expense_memo_structure.length > 0 ? this.advancedSettings?.expense_memo_structure : null],
+          topMemoStructure: [this.advancedSettings?.top_memo_structure.length > 0 ? this.advancedSettings?.top_memo_structure : null],
+          exportSchedule: [this.advancedSettings?.schedule_is_enabled ? true : false],
+          exportScheduleFrequency: [this.advancedSettings?.interval_hours ? this.advancedSettings?.interval_hours : null],
+          email: [this.advancedSettings?.emails.length > 0 ? this.advancedSettings?.emails : null],
+          frequency: [this.advancedSettings?.frequency ? this.advancedSettings?.frequency : null],
+          dayOfMonth: [this.advancedSettings?.day_of_month ? this.advancedSettings?.day_of_month : null],
+          dayOfWeek: [this.advancedSettings?.day_of_week ? this.advancedSettings?.day_of_week : null],
+          timeOfDay: [this.advancedSettings?.time_of_day ? this.advancedSettings?.time_of_day : null],
+          meridiem: [null]
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/advanced-setting/advanced-setting.component.ts:93]: post save how are we displaying this, why is this null?

File: src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.html [hunk 2]
```diff
@@ -1,16 +1,30 @@
...
+        <div class="tw-mr-20-px tw-pl-30-px tw-flex tw-justify-end tw-items-end" *ngIf="formControllerName === 'dayOfMonth' || formControllerName === 'dayOfWeek' || formControllerName === 'timeOfDay'">
+            <div *ngIf="formControllerName ===  'dayOfMonth' || formControllerName === 'dayOfWeek'" class="tw-mr-18-px dropDown-monthy">
+                <p class="tw-font-400 tw-text-faded-text-color tw-text-14-px">Day</p>
+                <p-dropdown [options]="options" [formControlName]="formControllerName" [placeholder]="placeholder"></p-dropdown>
+            </div>
+            <div class="tw-mr-18-px" [ngClass]="[formControllerName === 'dayOfMonth' || formControllerName === 'dayOfWeek' ? 'dropDown-time' : 'dropDown-time-alone']">
+                <p class="tw-font-400 tw-text-faded-text-color tw-text-14-px">Time</p>
+                <p-dropdown [options]="timeOption" formControlName="timeOfDay" placeholder="Select Time for export" ></p-dropdown>
+            </div>
+            <div [ngClass]="[formControllerName === 'dayOfMonth' || formControllerName === 'dayOfWeek' ? 'dropDown-time' : 'dropDown-time-alone']">
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.html:25]: `dropdown-time`

File: src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.ts [hunk 3]
```diff
@@ -23,8 +23,22 @@ export class ConfigurationSelectFieldComponent implements OnInit {
 
   @Input() formControllerName: string;
 
+
+  meridiemOption: string[] = ['AM', 'PM'];
+
+  timeOption: string[] = ["1:00", "1:30", "2:00", "2:30", "3:00", "3:30", "4:00", "4:30", "5:00", "5:30", "6:00", "6:30", "7:00", "7:30", "8:00", "8:30", "9:00", "9:30", "10:00", "10:30", "11:00", "11:30", "12:00", "12:30"];
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.ts:29]: single quotes for string, for consistency

File: src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.ts [hunk 4]
```diff
@@ -23,8 +23,22 @@ export class ConfigurationSelectFieldComponent implements OnInit {
...
+
+
   constructor() { }
 
   ngOnInit(): void {
-  }
+    if (this.formControllerName === 'dayOfMonth' || this.formControllerName === 'dayOfWeek' || this.formControllerName === 'timeOfDay') {
+      if (!this.form.value.meridiem) {
+        this.form.controls.meridiem.patchValue(this.meridiemOption[0]);
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.ts:37]: can we do this while we initialise form group?

File: src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.ts [hunk 5]
```diff
@@ -23,8 +23,22 @@ export class ConfigurationSelectFieldComponent implements OnInit {
...
+      }
+      if (!this.form.value.timeOfDay) {
+        this.form.controls.timeOfDay.patchValue(this.timeOption[0]);
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.ts:40]: same

File: src/app/shared/components/qbd/configuration/configuration-step-header/configuration-step-header.component.html [hunk 6]
```diff
@@ -2,8 +2,8 @@
     <h3 class="tw-text-normal-text-color tw-text-20-px tw-font-500">
         {{ headerText }}
     </h3>
-    <h5 class="tw-text-faded-text-color tw-text-14-px tw-pt-12-px">
-        {{ contentText }}
+    <h5 class="tw-text-faded-text-color tw-text-14-px tw-pt-12-px tw-flex">
+        {{ contentText }} &nbsp;
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/configuration-step-header/configuration-step-header.component.html:6]: what is this `&nbsp;`?

File: src/app/shared/components/qbd/configuration/email-multi-select-field/email-multi-select-field.component.html [hunk 7]
```diff
@@ -1 +1,83 @@
...
+            </div>
+
+            <div class="tw-flex tw-items-center tw-pt-24-px tw-pb-40-px">
+                <button pButton type="button" class="p-button-raised" [disabled]="!addEmailForm.valid" (click)="addEmail()">
+                    Save
+                </button>
+            </div>
+        </div>
+    </div>
+</p-dialog> -->
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/email-multi-select-field/email-multi-select-field.component.html:83]: why are these commented out?

File: src/styles.scss [hunk 8]
```diff
@@ -107,7 +107,7 @@ p {
 
 // Dropdown
 .p-dropdown {
-    @apply tw-w-80 #{!important};
+    @apply tw-w-80 ;
```
Comment (ashwin1111) [inline src/styles.scss:110]: make sure this doesn't affect bamboo

File: src/app/shared/components/qbd/configuration/configuration-step-header/configuration-step-header.component.html (hunk 6 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/configuration-step-header/configuration-step-header.component.html:6]: this is for space

File: src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.ts (hunk 4 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/configuration-select-field/configuration-select-field.component.ts:37]: we are doing this in email component, the option are in email component, so it can't do it in advanced setting

File: src/app/shared/components/qbd/configuration/email-multi-select-field/email-multi-select-field.component.ts [hunk 9]
```diff
@@ -7,9 +11,105 @@ import { Component, OnInit } from '@angular/core';
...
+
+  @Input() label: string;
+
+  @Input() subLabel: string;
+
+  @Input() placeholder: string;
+
+  @Input() formControllerName: string;
+
+  cities = [
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/email-multi-select-field/email-multi-select-field.component.ts:30]: why is this needed?

**Review by ashwin1111 — APPROVED** [review]
